File: 2215-find-the-difference-of-two-arrays/2215-find-the-difference-of-two-arrays.go
func findDifference(nums1 []int, nums2 []int) [][]int {
	var result [][]int
    set1 := map[int]bool{}
    set2 := map[int]bool{}

	for i := 0; i < len(nums1); i++ {
		set1[nums1[i]] = true
	}

	for i := 0; i < len(nums2); i++ {
		set2[nums2[i]] = true
	}

	result = append(result, []int{})

	for i := 0; i < len(nums1); i++ {
		if !set2[nums1[i]] {
			result[0] = append(result[0], nums1[i])
			set2[nums1[i]] = true
		}
	}

	result = append(result, []int{})
	
	for i := 0; i < len(nums2); i++ {
		if !set1[nums2[i]] {
			result[1] = append(result[1], nums2[i])
			set1[nums2[i]] = true
		}
	}
	return result
}

File: 2215-find-the-difference-of-two-arrays/2215-find-the-difference-of-two-arrays.java
class Solution {
    public List<List<Integer>> findDifference(int[] nums1, int[] nums2) {
        Set<Integer> set1 = new HashSet<>(Arrays.stream(nums1).boxed().collect(Collectors.toList()));
        Set<Integer> set2 = new HashSet<>(Arrays.stream(nums2).boxed().collect(Collectors.toList()));

        List<List<Integer>> list = new ArrayList<>();
        list.add(new ArrayList<>());

        for(int i=0; i<nums1.length; i++) {
            if(!set2.contains(nums1[i])) {
                list.get(0).add(nums1[i]);
                set2.add(nums1[i]);
            }
        }

        list.add(new ArrayList<>());

        for(int i=0; i<nums2.length; i++) {
            if(!set1.contains(nums2[i])) {
                list.get(1).add(nums2[i]);
                set1.add(nums2[i]);
            }
        }
        return list;
    }
}